lib/SlackEventHandler.js

```diff
@@ -140,13 +145,24 @@ SlackEventHandler.prototype.handleChannelRenameEvent = function (params) {
  * @param {string} params.event.channel The slack channel id
  * @param {string} params.event.ts The unique (per-channel) timestamp
  */
-SlackEventHandler.prototype.handleMessageEvent = function (params) {
+SlackEventHandler.prototype.handleMessageEvent = async function (params) {
     var room = this._main.getRoomBySlackChannelId(params.event.channel);
     if (!room) throw new UnknownChannel(params.event.channel);
 
     if (params.event.subtype === 'bot_message' &&
         (!room.getSlackBotId() || params.event.bot_id === room.getSlackBotId())) {
-        return Promise.resolve();
+        return;
+    }
+
+    // If it's a ephemeral event we don't have a way of relaying it.
+    if (params.event.is_ephemeral) {
+        return;
+    }
+
+    // If it's a message_deleted event we should redact it on the matrix side.
+    if (params.event.subtype === 'message_deleted') {
```

Comment:
Do we need this, @Cadair has a seperate PR for this so it feels like it's a pointless if loop